Userland/Games/GameOfLife/Pattern.cpp

```diff
@@ -24,13 +28,11 @@ void Pattern::set_action(GUI::Action* action)
 
 void Pattern::rotate_clockwise()
 {
-    Vector<DeprecatedString> rotated;
-    for (size_t i = 0; i < m_pattern.first().length(); i++) {
-        StringBuilder builder;
-        for (int j = m_pattern.size() - 1; j >= 0; j--) {
-            builder.append(m_pattern.at(j).substring(i, 1));
-        }
-        rotated.append(builder.to_deprecated_string());
-    }
+    Vector<char> rotated;
```

Comment:
Looks like you missed out `rotated.resize(m_pattern.size())` here

Reply:
Added the resizing here.
Does this really matter at this method? Rotation currently only happens when the user clicks on the "rotate pattern" button in the toolbar (see main.cpp:129-133). The performance gain achieved by this might be unnoticeable to the user as long as only a single pattern gets rotated.

Reply:
If you index into a zero length vector the code will crash I expect. Did you test this?

Reply:
Ah that answers my question ... I was thinking the vector would resize itself automatically - but this wont happen when (new) elements are set by index. I will correct that.